File: VB-Examples/11_Formatting/GetConditionalFormatColor.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text
Imports Spire.Xls.Core.Spreadsheet

Namespace GetConditionalFormatColor
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load a workbook from the specified file path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\Template_Xls_13.xlsx")

            ' Get the first worksheet from the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Define a CellRange object representing range A1:C1 in the worksheet
            Dim cRange As CellRange = sheet.Range("A1:C1")

            ' Get the color of the condition format applied to the range
            Dim color = cRange.GetConditionFormatsStyle().Color

            ' Display a message box showing the color of the condition format
            MessageBox.Show("The color of the condition format is " & color.ToString())

		End Sub


		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/11_Formatting/GetConditionalFormatColor.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Xls.Core.Spreadsheet;

namespace GetConditionalFormatColor
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a new workbook object
            Workbook workbook = new Workbook();

            // Load an existing Excel document
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\Template_Xls_13.xlsx");

            // Get the first sheet from the workbook
            Worksheet sheet = workbook.Worksheets[0];

            // Define a cell range
            CellRange cRange = sheet.Range["A1:C1"];

            // Retrieve the color of the condition format applied to the cell range
            var color = cRange.GetConditionFormatsStyle().Color;

            // Display a message box with the color information
            MessageBox.Show("The color of the condition format is " + color.ToString());

            // Dispose of the workbook object to release resources
            workbook.Dispose();
        }


        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
